superset/views/core.py

```diff
@@ -3093,26 +3092,38 @@ def sqllab(self):
                 }
                 for database in db.session.query(models.Database).all()
             }
+            # return all user queries associated with existing SQL editors
             user_queries = (
-                db.session.query(Query).filter_by(user_id=g.user.get_id()).all()
+                db.session.query(Query)
+                .filter_by(user_id=user_id)
+                .filter(Query.sql_editor_id.cast(Integer).in_(tab_state_ids))
```

Comment:
is this not an int by default?

Reply:
No, this is a string because in the pre-`SQLLAB_BACKEND_PERSISTENCE` world it's generated randomly.

Reply:
@betodealmeida we're seeing the following error due to the casting as an Integer:
```
sqlalchemy.exc.DataError: (psycopg2.DataError) invalid input syntax for integer: "dKvEsdZdq"
```
Any reason why `tab_state.id` should not be a `varchar` to match `query.sql_editor_id`?

Reply:
@robdiciuccio with the persistence feature flag enabled the ID for each tab is autogenerated from the autoincrement key, so it's an int. But it could be a string instead, autogenerated in the backend. I opted for an int because it seems like a better solution for the long run, and tried to make the cast work everywhere.